Refactor find command output to ripgrep-style format with colors

estyxx opened the pull request:
This PR refactors the `find` command output to match ripgrep's familiar format, making results more compact.

## Changes

- **Output format**: Changed from styled snippet annotations to ripgrep-style grouped-by-file format
- **Color coding**: Added terminal colors matching ripgrep's color scheme:
  - File paths in magenta/pink
  - Line numbers in bright green  
  - Matched text highlighted in red
- **Grouping**: Results are now grouped by file, with all matches for a file shown together

## Before
<img width="1680" height="1584" alt="CleanShot 2025-10-31 at 15 14 02@2x" src="https://github.com/user-attachments/assets/487893ae-4703-4912-bad7-2f4d2c1809bf" />

## After 

<img width="1656" height="978" alt="CleanShot 2025-11-01 at 14 39 58@2x" src="https://github.com/user-attachments/assets/d9f25dbc-6504-403f-b80b-8e503ceaa2aa" />

estyxx committed:
Refactor find output to ripgrep-style format grouped by file

estyxx committed:
Add colored output: magenta file paths, green line numbers, red matches

estyxx committed:
Update README.md to reflect new ouput format

feds01 commented on crates/teamsearch/src/lib.rs:
```diff
@@ -91,6 +92,68 @@ fn resolve_default_files(files: Vec<PathBuf>, is_stdin: bool) -> Vec<PathBuf> {
+/// Get the line number (1-indexed) and byte range for a given byte position.
+///
+/// Returns a tuple of (line_number, line_start_byte, line_end_byte) where line_number is 1-indexed.
+fn get_line_range(contents: &str, byte_pos: usize) -> (usize, usize, usize) {
```
I think we can move this into `crates/teamsearch_utils/src/lines.rs`

feds01 commented on crates/teamsearch/src/lib.rs:
```diff
@@ -116,28 +179,53 @@ fn find(args: FindCommand) -> Result<ExitStatus> {
-                let mut message = Level::Info.title("match found");
-
-                // Now, construct the reports so that we can emit them to the user.
+                use colored::Colorize;
```
This should be at the crate level.

feds01 commented on crates/teamsearch/src/lib.rs:
```diff
@@ -91,6 +92,68 @@ fn resolve_default_files(files: Vec<PathBuf>, is_stdin: bool) -> Vec<PathBuf> {
+
+/// Highlight matches in a line of text.
+fn highlight_line_matches(line_content: &str, matches: &[Match]) -> String {
+    use colored::Colorize;
```
♻️

feds01 commented on crates/teamsearch/src/lib.rs:
```diff
@@ -91,6 +92,68 @@ fn resolve_default_files(files: Vec<PathBuf>, is_stdin: bool) -> Vec<PathBuf> {
+}
+
+/// Highlight matches in a line of text.
+fn highlight_line_matches(line_content: &str, matches: &[Match]) -> String {
```
I think we can use `highlight` [^1]?

[^1]: https://github.com/feds01/teamsearch/blob/7d86d1c1169b6b5f5ef8ac39c1572a0f537ba691/crates/teamsearch_utils/src/highlight.rs#L99

estyxx replied:
Nice! done it!

feds01 reviewed (commented):
Thank you for the contribution ❤️  I think this is a nice improvement over the existing display of results. Happy to accept this, I've left a couple of nits on the PR. Once addresses, I think this can go in!

estyxx committed:
Add utility to get line number and byte range for given position

Introduce a new function `get_line_range` in the `teamsearch_utils` crate for
parsing line numbers and byte ranges from a string, given a byte position.

feds01 reviewed (approved):
Looks great! Thank you ❤️
